lib/Graph/Graph.cpp

```diff
@@ -2137,20 +2155,21 @@ void Function::verify() const {
     llvm_unreachable("Multiple nodes with the same name");
   }
 
-  const auto &vars = getParent()->getVars();
-  (void)vars;
-
   // Any node referenced by one of the graph nodes should be part of the Graph.
   for (const auto &N : nodes_) {
     for (size_t idx = 0, e = N.getNumInputs(); idx < e; ++idx) {
       auto &input = N.getNthInput(idx);
       (void)input;
       // Verify each input of N.
       verifyNodeInput(N, idx);
-      assert((std::find(nodes_.begin(), nodes_.end(), *input) != nodes_.end() ||
-              std::find(vars.begin(), vars.end(), input) != vars.end()) &&
-             "Every node referenced by one of the graph"
-             " nodes should be part of the graph");
+      bool foundNode =
+          std::find(nodes_.begin(), nodes_.end(), *input) != nodes_.end();
+      bool foundStorage = isGraphStorageNode(input, this);
```

Comment:
nit: this isn't necessary to run if `foundNode = true`, maybe instead consider `if (!foundNode) foundNode = isGraphStorageNode()`.